tamoc/ambient.py

```diff
@@ -178,7 +182,14 @@ def _create_profile_from_xarray(self, time=None):
             if fs_loc > 0:
                 fs_loc = -1
             # Compute the pressure for this density profile
-            Ps = compute_pressure(zs, Ts, Ss, fs_loc)
+            if self.ztsp_units[1] != "K":
+                # Note -- this should be done by xr_convert_units, but
+                # that's expecting pressure to be there already.
+                T_K, units = convert_units(np.array(Ts.reshape((-1, 1))),
+                                           [self.ztsp_units[1]])
+            else:
+                T_K = Ts
+            Ps = compute_pressure(zs, T_K, Ss, fs_loc)
```

Comment:
This is so it will use the right units of temp before computing the pressure profile -- it would problaby be better to normalize all units first, but the code for that currently expects pressure to be there already, so I hacked this in.